### PR TITLE
feat(results,schema): CPU microarch fingerprint + forensics schema (ENG-70)

### PR DESCRIPTION
**Pull request by dbworku**
**ENG-70 host fingerprint — split into 3 focused PRs** (merge bottom-up, each independently green):
1. **#82 — CPU microarch fingerprint + forensics schema** (this PR)
2. #83 — effective-spec probe parser + fixtures
3. #72 — host-fingerprint enrichment onto the tree

↑ **This PR is 1/3**, based on #81.

---
**Leaves + schema plumbing**: the pure CPU-microarch fingerprint resolver, the PTS `System` type, the observedSpecs `cpuMicroarch`/`hostHeterogeneous` schema fields, the `ptsForensics` raw-file recognizer, and the `bench.sh` forensics-tarball step. All pure-additive; the probe parser and wiring that consume them land in #83 and #72.

**Validated locally:** `bun run typecheck` (8 workspaces, 0 errors); `@sandbox-benchmarks/results` 41 and `@sandbox-benchmarks/schema` 133 tests pass.

---
## Summary by cubic
Adds a host CPU microarchitecture fingerprint resolver, PTS `<System>` parsing, and a `pts_<test>--forensics.tar.gz` artifact for ENG-70. Host-only leaves; probe parsing and wiring land in follow-ups.

- **New Features**
  - `packages/results`: `resolveCpuMicroarch` resolves via family/model table, EPYC/Xeon SKU tables (incl. AWS/Azure masked SKUs), and AMD AVX‑512 inference.
  - `packages/results`: Parse PTS `<System>` in `ptsCompositeSchema`; export `PtsSystem`.
  - `packages/schema`: `ObservedSpecs` adds `cpuMicroarch?` (host-only) and `hostCpuModels?` (aggregate-only distinct host CPU models).
  - `lib/bench.sh` + `packages/schema`: Emit `pts_<test>--forensics.tar.gz`; add `isPtsForensicsFile`/`ptsForensicsFile`.

- **Bug Fixes**
  - Forensics tarball creation is best-effort (`|| true`) and won’t fail a run.
  - Forensics tarball names never match `.xml` and require a non-empty `<test>` segment.

<sup>Written for commit e7ed6c1a79b3cecf6cf795bcd5272d9dca00d9e1. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/starslingdev/sandbox-benchmarks/pull/82?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Added host CPU microarchitecture and heterogeneous-host signaling to observed benchmark specs.
  * Added host CPU fingerprint resolution for more AMD EPYC and Intel Xeon formats, including inference from ISA hints.
  * Parse optional host `<System>` data from PTS composite results and expose it in public types.
  * Benchmarks now generate per-run `--forensics.tar.gz` archives; raw artifact detection supports them.
* **Bug Fixes**
  * Forensics archive creation failures no longer break benchmarks.
* **Tests**
  * Added coverage for CPU resolution and PTS forensics raw-file naming edge cases.

### CODE REVIEW

**Comment by dbworku** [comment]
* **#73** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/73?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#74](https://github.com/starslingdev/sandbox-benchmarks/pull/74) <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/74?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#75](https://github.com/starslingdev/sandbox-benchmarks/pull/75) <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/75?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#72** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/72?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#83** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/83?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#82** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/82?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/82?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#81** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/81?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#71** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/71?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#80** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/80?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#68** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/68?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#84](https://github.com/starslingdev/sandbox-benchmarks/pull/84) <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/84?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#79** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/79?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#67** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/67?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#78** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/78?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#88** <a href="https://app.graphite.com/github/pr/starslingdev/sandbox-benchmarks/88?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>merge-ready</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/starslingdev?ref=merge-queue-instructions-in-stack-comment&prId=82).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/results/src/lib/cpu-fingerprint.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	return undefined;
+}
+
+/** Map an Intel Xeon brand string to a microarch via the SKUs seen across the runner composites. */
+function resolveXeon(brand: string): CpuFingerprint | undefined {
+	if (/Platinum 8488C/.test(brand)) return { name: "Sapphire Rapids", year: 2023 };
+	if (/Platinum 8375C/.test(brand)) return { name: "Ice Lake-SP", year: 2021 };
+	if (/Platinum 8358/.test(brand) || /Platinum 8370C/.test(brand)) {
+		return { name: "Ice Lake-SP", year: 2021 };
+	}
```
[the remark this answers is not in the document]
Comment (dbworku) [reply packages/results/src/lib/cpu-fingerprint.ts:83]: Keeping the explicit per-SKU Xeon checks — they read as a lookup table and are trivial to extend per observed composite; a combined regex is terser but less greppable.

File: lib/bench.sh [hunk 2]
```diff
@@ -242,6 +242,14 @@ run_pts_benchmark() {
...
 		cp "$xml_found" "$(results_dir)/${prefix}.xml" 2>/dev/null || true
 		echo "Structured result: ${prefix}.xml (from $(dirname "$xml_found"))"
+		# Capture the whole result dir (composite.xml + installation-logs/ + test-logs/) as a forensics
+		# tarball for debugging. A .tar.gz — not a flattened copy — so its nested .xml files can't be
+		# misrouted by the extractor (the name ends --forensics.tar.gz, which isPtsResultFile never
+		# matches). `|| true` so a /var/lib perms hiccup can't abort this `set -e` measurement leaf.
+		local result_dir
+		result_dir="$(dirname "$xml_found")"
+		tar -czf "$(results_dir)/${prefix}--forensics.tar.gz" \
+			-C "$(dirname "$result_dir")" "$(basename "$result_dir")" 2>/dev/null || true
```
[the remark this answers is not in the document]
Comment (dbworku) [reply lib/bench.sh:252]: Micro-optimization in a once-per-step path; not worth trading the current readability.

[the remark this answers is not in the document]
Comment (dbworku) [reply lib/bench.sh:252]: The suffix is necessarily duplicated across the shell producer and the TS recognizer (bash can't import the TS constant). Both sides are tested and raw-files.test pins the contract, so drift would fail CI.

File: packages/results/src/lib/cpu-fingerprint.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	// 9xx5 = Turin (Zen 5), 9xx4 = Genoa/Bergamo (Zen 4), 7xx3 = Milan (Zen 3), 7xx2 = Rome (Zen 2).
+	const m = brand.match(/EPYC\s+(\d)\d{2}(\d)/);
+	if (m) {
+		const series = m[1];
+		const gen = m[2];
+		if (series === "9" && gen === "5") return { name: "Zen 5 (Turin)", year: 2024 };
+		if (series === "9" && gen === "4") return { name: "Zen 4 (Genoa)", year: 2022 };
+		if (series === "7" && gen === "3") return { name: "Zen 3 (Milan)", year: 2021 };
+		if (series === "7" && gen === "2") return { name: "Zen 2 (Rome)", year: 2019 };
+	}
```
[the remark this answers is not in the document]
Comment (dbworku) [reply packages/results/src/lib/cpu-fingerprint.ts]: Fixed: the EPYC regex now allows a letter in the second position ((\d)[A-Z\d]\d(\d)) so 7F32/7F72/7H12 resolve to Zen 2 (Rome).

File: packages/results/src/lib/cpu-fingerprint.test.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	it("resolves standard EPYC SKUs by their model number (the final digit is the generation)", () => {
+		// OURS fixtures (pts_stream/pts_hardlink) run on a bare-metal 9275F — a 9xx5 = Zen 5 (Turin).
+		expect(resolveCpuMicroarch("AMD EPYC 9275F 24-Core")).toEqual({
+			name: "Zen 5 (Turin)",
+			year: 2024,
+		});
+		// 9xx4 = Genoa (Zen 4), 7xx3 = Milan (Zen 3), 7xx2 = Rome (Zen 2).
+		expect(resolveCpuMicroarch("AMD EPYC 9654 96-Core Processor")?.name).toBe("Zen 4 (Genoa)");
+		expect(resolveCpuMicroarch("AMD EPYC 7763 64-Core Processor")?.name).toBe("Zen 3 (Milan)");
+	});
```
[the remark this answers is not in the document]
Comment (dbworku) [reply packages/results/src/lib/cpu-fingerprint.test.ts:19]: Fixed: added 7F32/7F72/7H12 → Zen 2 (Rome) test cases.

File: packages/schema/src/raw-files.ts [hunk 5]
```diff
@@ -28,12 +28,30 @@ const SKIP_SUFFIX = "--skipped.json";
...
+ * nested `.xml` files can't be misrouted by {@link isPtsResultFile}: it `startsWith("pts_")` yet ends
+ * `--forensics.tar.gz`, so it is provably disjoint from the `.xml` result predicate.
+ */
+export function ptsForensicsFile(prefix: string): string {
+	return `${prefix}${FORENSICS_SUFFIX}`;
+}
+
+const ptsForensicsFileName = type("string").matching("^pts_.*--forensics\\.tar\\.gz$");
+export const isPtsForensicsFile = (filename: string): boolean =>
+	ptsForensicsFileName.allows(filename);
```
[the remark this answers is not in the document]
Comment (dbworku) [reply packages/schema/src/raw-files.ts]: Fixed: tightened the forensics pattern from .* to .+ so an empty profile segment (pts_--forensics.tar.gz) is rejected.

File: packages/schema/src/raw-files.test.ts [hunk 6]
```diff
@@ -16,6 +18,16 @@ describe("raw-file naming", () => {
...
 
+	it("names a forensics tarball and keeps it disjoint from the PTS result predicate", () => {
+		const file = ptsForensicsFile("pts_node-web-tooling");
+		expect(file).toBe("pts_node-web-tooling--forensics.tar.gz");
+		expect(isPtsForensicsFile(file)).toBe(true);
+		// Provably disjoint: the tarball starts pts_ but must NEVER route through the .xml extractor.
+		expect(isPtsResultFile(file)).toBe(false);
+		// And a real result XML is not a forensics tarball.
+		expect(isPtsForensicsFile("pts_node-web-tooling.xml")).toBe(false);
+	});
```
[the remark this answers is not in the document]
Comment (dbworku) [reply packages/schema/src/raw-files.test.ts:31]: Fixed: added an assertion that pts_--forensics.tar.gz is rejected.

**Review by cosmic-emerald — APPROVED** [review]
